bin/fetch_gha_min_cost.py

```diff
@@ -0,0 +1,103 @@
+import os
+import json
+from datetime import datetime,timezone, timedelta
+import concurrent.futures
+from services.github_service import GithubService
+
+
+def get_environment_variables() -> str:
+    github_token = os.getenv("ADMIN_GITHUB_TOKEN")
+    if not github_token:
+        raise ValueError(
+            "The env variable ADMIN_GITHUB_TOKEN is empty or missing")
+
+    return github_token
+
+
+def calculate_gha_cost_per_run(run_id, repo_object, github_service, os_multipliers, minute_cost_usd):
+
+    total_minutes = 0
+    cost_per_run = 0
+    response = github_service.get_workflow_run_details(repo_name=repo_object.full_name, run_id=run_id)
+
+    for os_type, multiplier in os_multipliers.items():
+        billable_data = response["billable"].get(os_type)
+        if billable_data:
+            total_ms = billable_data['total_ms']
+            total_minutes += total_ms / 60000
+            cost_per_run += total_minutes * minute_cost_usd * multiplier
+
+    return total_minutes, cost_per_run
+
+
+def calculcate_gha_cost_per_repo(repo_object, start_date, end_date, github_service, os_multipliers, minute_cost_usd):
+
+    total_minutes_repo = 0
+    cost_repo = 0
+    repo_workflow_runs = github_service.get_workflow_runs_repo(repo=repo_object, created=f"{start_date}..{end_date}")
+
+    for run in repo_workflow_runs: 
+        total_minutes_run, cost_per_run = calculate_gha_cost_per_run(run.id, repo_object, github_service, os_multipliers, minute_cost_usd)
+        if total_minutes_run > 0: 
+            total_minutes_repo = total_minutes_repo + total_minutes_run
+            cost_repo = round((cost_repo + cost_per_run), 3)
+
+    return total_minutes_repo, cost_repo
+
+
+def process_repository(repo_object, start_date, end_date, github_service, os_multipliers, minute_cost_usd): 
+
+    workflows = github_service.get_workflows_per_repo(repo=repo_object)
+    if workflows.totalCount > 0:
+        total_minutes_repo, cost_repo = calculcate_gha_cost_per_repo(
+            repo_object, start_date, end_date,
+            github_service, os_multipliers, minute_cost_usd)
+
+        if cost_repo > 0:
+            return {
+                "date": end_date,
+                "repo_name": repo_object.name,
+                "gha_minutes_usage_min": total_minutes_repo,
+                "gha_minutes_cost_usd": cost_repo,
+            }
+
+    return None
+
+
+def main(org_name: str = "ministryofjustice",
```

Comment:
Don't we want to do this for moj-analytical-services organisation as-well?

Reply:
@levgorbunov1 good point. The data fetching scripts needs to be run separately for each organization. So I was think it needs to be passed as a command-line arguments inside the workflow. What do you think?

Reply:
like this https://github.com/ministryofjustice/operations-engineering/blob/394fd07661a6e63e3939173eda8dfaf15493160d/bin/identify_dormant_github_users_v2.py#L110